model/task/task_test.go

```diff
@@ -1241,6 +1242,32 @@ func TestUnscheduleStaleUnderwaterTasks(t *testing.T) {
 	assert.EqualValues(-1, dbTask.Priority)
 }
 
+func TestUnscheduleStaleUnderwaterTasksWithDistro(t *testing.T) {
+	assert.NoError(t, db.ClearCollections(Collection, distro.Collection))
+	t1 := Task{
+		Id:            "t1",
+		Status:        evergreen.TaskUndispatched,
+		Activated:     true,
+		Priority:      0,
+		ActivatedTime: time.Time{},
+		DistroId:      "d0",
+	}
+	require.NoError(t, t1.Insert())
+
+	d := distro.Distro{
+		Id:      "d0",
+		Aliases: []string{"d0.0", "d0.1"},
+	}
+	require.NoError(t, d.Insert())
+
+	_, err := UnscheduleStaleUnderwaterTasks("d0")
+	assert.NoError(t, err)
+	dbTask, err := FindOneId("t1")
+	assert.NoError(t, err)
+	assert.False(t, dbTask.Activated)
+	assert.EqualValues(t, -1, dbTask.Priority)
```

Comment:
Can we add a test for the case of a task with the deleted + aliased distro as well?

Reply:
done